nodejs/lib/inference-session.ts

```diff
@@ -0,0 +1,242 @@
+   */
+  readonly inputNames: readonly string[];
+
+  /**
+   * Get output names of the loaded model.
+   */
+  readonly outputNames: readonly string[];
+
+  // /**
+  //  * Get input metadata of the loaded model.
+  //  */
+  // readonly inputMetadata: ReadonlyArray<Readonly<InferenceSession.ValueMetadata>>;
+
+  // /**
+  //  * Get output metadata of the loaded model.
+  //  */
+  // readonly outputMetadata: ReadonlyArray<Readonly<InferenceSession.ValueMetadata>>;
+
+  //#endregion
+}
+
+export declare namespace InferenceSession {
+  //#region input/output types
+
+  type OnnxValueMapType = {readonly [name: string]: OnnxValue};
+  type NullableOnnxValueMapType = {readonly [name: string]: OnnxValue | null};
+
+  /**
+   * A feeds (model inputs) is an object that use input names as keys and OnnxValue as corresponding values.
+   */
+  type FeedsType = OnnxValueMapType;
+
+  /**
+   * A fetches (model outputs) could be one of the following:
+   *
+   * - Omitted. Use model's output names definition.
+   * - An array of string indicating the output names.
+   * - An object that use output names as keys and OnnxValue or null as corresponding values.
+   *
+   * REMARK: different from input argument, in output, OnnxValue is optional. If an OnnxValue is present it will be
+   * used as a pre-allocated value by the inference engine; if omitted, inference engine will allocate buffer
+   * internally.
+   */
+  type FetchesType = readonly string[]|NullableOnnxValueMapType;
+
+  type ReturnType = {readonly [name: string]: OnnxValue};
+
+  //#endregion
+
+  //#region session options
+
+  /**
+   * A set of configurations for session behavior.
+   */
+  export interface SessionOptions {
+    /**
+     * An array of execution provider options.
+     *
+     * An execution provider option can be a string indicating the name of the execution provider,
+     * or an object of corresponding type.
+     */
+    executionProviders?: readonly SessionOptions.ExecutionProviderConfig[];
+
+    /**
+     * The intra OP threads number.
+     */
+    intraOpNumThreads?: number;
+
+    /**
+     * The inter OP threads number.
+     */
+    interOpNumThreads?: number;
+
+    /**
+     * The optimization level.
+     */
+    graphOptimizationLevel?: 'disabled'|'basic'|'extended'|'all';
+
+    /**
+     * Whether enable CPU memory arena.
+     */
+    enableCpuMemArena?: boolean;
+
+    /**
+     * Whether enable memory pattern.
+     */
+    enableMemPattern?: boolean;
+
+    /**
+     * Execution mode.
+     */
+    executionMode?: 'sequential'|'parallel';
+
+    /**
+     * Log ID.
+     */
+    logId?: string;
+
+    /**
+     * Log severity level. See
+     * https://github.com/microsoft/onnxruntime/blob/master/include/onnxruntime/core/common/logging/severity.h
+     */
+    logSeverityLevel?: 0|1|2|3|4;
+  }
+
+  export namespace SessionOptions {
+    //#region execution providers
+
+    // currently we only have CPU, DNNL and CUDA EP. in future to support more.
```

Comment:
Is DNNL an EP or just a build flavor of the CPU build? The comment seems to indicate 3 EPs but only 2 reside in the map below.

Reply:
This is my bad. I have removed DNNL support but I didn't clean up the comments. will remove from the comments.